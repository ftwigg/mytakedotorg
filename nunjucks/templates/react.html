{% extends "reactCore.html" %}

{% block bundle %}
	<!-- Main -->
	<script src="/app.bundle.js"></script>
{% endblock %}
