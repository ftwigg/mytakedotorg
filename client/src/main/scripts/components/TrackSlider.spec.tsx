import * as React from "react";
import * as renderer from "react-test-renderer";
import {} from "jest";
import TrackSlider, { TrackSliderEventHandlers } from "./TrackSlider";
import {
  initialRangeSliders,
  unzoomedRangeSliders,
  zoomedRangeSliders
} from "../utils/testUtils";

const eventHandlers: TrackSliderEventHandlers = {
  onAfterRangeChange: jest.fn(),
  onRangeChange: jest.fn()
};

test("Initial render", () => {
  const tree = renderer
    .create(
      <TrackSlider
        start={0}
        end={5224}
        eventHandlers={eventHandlers}
        rangeSliders={initialRangeSliders}
        step={1}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Zoomed render", () => {
  const tree = renderer
    .create(
      <TrackSlider
        start={4}
        end={82}
        eventHandlers={eventHandlers}
        rangeSliders={zoomedRangeSliders}
        step={0.1}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Unzoomed render", () => {
  const tree = renderer
    .create(
      <TrackSlider
        start={4}
        end={82}
        eventHandlers={eventHandlers}
        rangeSliders={unzoomedRangeSliders}
        step={0.1}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
