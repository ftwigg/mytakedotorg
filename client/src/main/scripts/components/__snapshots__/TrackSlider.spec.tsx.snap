// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial render 1`] = `
<div
  className="trackSlider"
>
  <p
    className="trackSlider__text trackSlider__text--min"
  >
    0
  </p>
  <div
    className="trackSlider__range-container"
  >
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0.019142419601837674%",
              "visibility": "visible",
              "width": "0.32542113323124044%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={5224}
          aria-valuemin={0}
          aria-valuenow={1}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0.019142419601837674%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={5224}
          aria-valuemin={0}
          aria-valuenow={18}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0.3445635528330781%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Transcript
      </span>
    </div>
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0%",
              "visibility": "visible",
              "width": "0%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={5224}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={5224}
          aria-valuemin={0}
          aria-valuenow={0}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Clip
      </span>
    </div>
  </div>
  <p
    className="trackSlider__text trackSlider__text--max"
  >
    1:27:04
  </p>
</div>
`;

exports[`Unzoomed render 1`] = `
<div
  className="trackSlider"
>
  <p
    className="trackSlider__text trackSlider__text--min"
  >
    4
  </p>
  <div
    className="trackSlider__range-container"
  >
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0%",
              "visibility": "visible",
              "width": "31.025641025641026%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={4}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={28.2}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "31.025641025641026%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Transcript
      </span>
    </div>
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0%",
              "visibility": "visible",
              "width": "1.4102564102564097%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={4}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={5.1}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "1.4102564102564097%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Clip
      </span>
    </div>
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0%",
              "visibility": "visible",
              "width": "100%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={4}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={82}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "100%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Zoom
      </span>
    </div>
  </div>
  <p
    className="trackSlider__text trackSlider__text--max"
  >
    1:22
  </p>
</div>
`;

exports[`Zoomed render 1`] = `
<div
  className="trackSlider"
>
  <p
    className="trackSlider__text trackSlider__text--min"
  >
    4
  </p>
  <div
    className="trackSlider__range-container"
  >
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0%",
              "visibility": "visible",
              "width": "31.025641025641026%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={4}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={28.2}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "31.025641025641026%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Transcript
      </span>
    </div>
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0%",
              "visibility": "visible",
              "width": "1.4102564102564097%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={4}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={5.1}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "1.4102564102564097%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Clip
      </span>
    </div>
    <div
      className="trackSlider__range"
    >
      <div
        className="rc-slider"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <div
          className="rc-slider-rail"
          style={
            Object {
              "backgroundColor": "#d3dae3",
            }
          }
        />
        <div
          className="rc-slider-track rc-slider-track-1"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "left": "0%",
              "visibility": "visible",
              "width": "1.5384615384615388%",
            }
          }
        />
        <div
          className="rc-slider-step"
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={4}
          className="rc-slider-handle rc-slider-handle-1"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "0%",
            }
          }
          tabIndex={0}
        />
        <div
          aria-disabled={false}
          aria-valuemax={82}
          aria-valuemin={4}
          aria-valuenow={5.2}
          className="rc-slider-handle rc-slider-handle-2"
          role="slider"
          style={
            Object {
              "backgroundColor": "#758aa8",
              "border": "1px solid #2c4770",
              "left": "1.5384615384615388%",
            }
          }
          tabIndex={0}
        />
        <div
          className="rc-slider-mark"
        />
      </div>
      <span
        className="trackSlider__label"
      >
        Zoom
      </span>
    </div>
  </div>
  <p
    className="trackSlider__text trackSlider__text--max"
  >
    1:22
  </p>
</div>
`;
